docs/program-setup/bare_metal/mbed2_porting.md

```diff
@@ -0,0 +1,125 @@
+
+Configure your target to support Mbed OS 6:
+- Remove the `release_versions` property as it is no longer required.
+```
+"release_versions": ["2"]
+```
+
+- Add the `supported_application_profiles` property to indicate that the application profile supported by this target is bare metal.
+```
+"supported_application_profiles" : ["bare-metal"]
+```
+
+- Override `supported_c_libs` property to link with the smaller C libraries. The default for all targets is defined as follows: 
+```
+"supported_c_libs": {
+    "arm":  ["std"],
+    "gcc_arm":  ["std", "small"],
+    "iar": ["std"]
+}
+```
+
+Both the ARM and GCC_ARM toolchains support optimized versions of the C standard libraries, microlib and newlib-nano respectively. We recommend using them with the bare metal profile to achieve lower memory footprints. Ultra constrained targets should override `supported_c_libs` as follows:
+```
+"supported_c_libs": {
+    "arm":  ["small"],
+    "gcc_arm":  ["small"]
+}
+```
+
+For each toolchain, if there is enough memory to link with the standard library, add the corresponding "std" library to the list. For example:
+```
+"supported_c_libs": {
+    "arm":  ["std", "small"],
+    "gcc_arm":  ["std", "small"],
+    "iar": ["std"]
+}
+```
+<span class="notes">**Note:** For ARM, your target scatter file needs to have a two-region memory model. If the build system throws an error (presence of undefined symbols `Image$$ARM_LIB_HEAP$$ZI$$Base`, `Image$$ARM_LIB_HEAP$$ZI$$Length`, `Image$$ARM_LIB_HEAP$$ZI$$Limit`) when compiling with microlib, you can find more information [here](https://os.mbed.com/docs/mbed-os/v6.0-preview/reference/using-small-c-libraries.html).</span>
+
+<span class="notes">**Note:** The IAR toolchain does not have a small C library.</span>
+
+- Set the `c_lib` property to indicate which C library should be used by the build tools. If your target has `"default_lib": "small"` defined, then replace it with `"c_lib" : "small"`. Otherwise add `"c_lib" : "small"`. We recommend that ultra constrained devices running bare metal applications link with the small C libraries by default.
+
+<span class="notes">**Note:** If a toolchain does not support a small C library and `c_lib` is set to `small`, the build tools will revert to linking with the standard C library, provided that it is listed in `supported_c_libs` for the given toolchain.</span>
+
+- If `default_toolchain` is set to `uARM`, then replace it with `ARM` and remove `uARM` from `supported_toolchains`. Support for the uARM toolchain, which is the ARM toolchain with microlib, has been removed. Setting `c_lib` to `small` ensures that microlib is used when the ARM toolchain is selected.
+
+<span class="notes">**Note:** As mentioned above, to successfully build with microlib, the target must define a two-region memory model. You might need to replace the content of the TOOLCHAIN_ARM_STD linker file with that of TOOLCHAIN_ARM_MICRO which includes a two-region memory model linker file.</span>
```

Comment:
```suggestion
<span class="notes">**Note:** As mentioned above, to successfully build with microlib, the target must define a two-region memory model. You might need to replace the content of the `TOOLCHAIN_ARM_STD` linker file with that of `TOOLCHAIN_ARM_MICRO`, which includes a two-region memory model linker file.</span>
```

Reply:
Question: under what circumstances will I need to do this?

Reply:
You will need to do this if there is a compilation error as mentioned above (line 62). This is a bit of a repeat and could be combined with the note above.